execution-core/src/transfer.rs

```diff
@@ -505,20 +508,224 @@ pub struct PhoenixTransactionEvent {
 }
 
 /// Event data emitted on a moonlight transaction's completion.
+#[cfg_attr(feature = "serde_support", derive(SerdeSerialize, SerdeDeserialize))]
 #[derive(Debug, Clone, Archive, PartialEq, Serialize, Deserialize)]
 #[archive_attr(derive(CheckBytes))]
 pub struct MoonlightTransactionEvent {
     /// The account that initiated the transaction.
+    #[cfg_attr(feature = "serde_support", serde(with = "serde_support::pubk"))]
     pub sender: AccountPublicKey,
     /// The receiver of the funds if any were transferred.
+    #[cfg_attr(
+        feature = "serde_support",
+        serde(with = "serde_support::optional_pubk")
+    )]
     pub receiver: Option<AccountPublicKey>,
     /// Transfer amount
     pub value: u64,
     /// The memo included in the transaction.
+    #[cfg_attr(
+        feature = "serde_support",
+        serde(with = "serde_support::bs58_serde")
+    )]
     pub memo: Vec<u8>,
     /// Gas spent by the transaction.
     pub gas_spent: u64,
     /// Optional refund-info in the case that the refund-address is different
     /// from the sender.
+    #[cfg_attr(
+        feature = "serde_support",
+        serde(with = "serde_support::pubk_u64_tuple")
+    )]
     pub refund_info: Option<(AccountPublicKey, u64)>,
 }
+
+#[cfg(feature = "serde_support")]
+mod serde_support {
```

Comment:
It probably makes sense to have this as a separate file. Then you can also import the modules in it that are likely to be reused (e.g., bs58_serde) on other structs located in other files.